js/experiment.js

```diff
@@ -168,75 +187,183 @@ function nodeClicked(e) {
-            curPath += 'L' + (parseFloat(e.currentTarget.getAttribute('x')) + nodeWidth/2 /*- 10*/) + 
-                        ',' + (parseFloat(e.currentTarget.getAttribute('y')) /*- 10*/);  
+            if (curPath === null) {
+                // later we can call findClosest with 'node', 'node'
+                /*curPath = 'M' + (parseFloat(startNode.getAttribute('x')) + nodeWidth/2) + 
+                                ',' + parseFloat(startNode.getAttribute('y')) +
+                    ' L' + (parseFloat(e.currentTarget.getAttribute('x')) + nodeWidth/2) + 
+                        ',' + (parseFloat(e.currentTarget.getAttribute('y')));*/
+                findClosest( {x: parseFloat(startNode.getAttribute('x')), 
+                              y: parseFloat(startNode.getAttribute('y'))},
+                            'node', 
+                            {x: parseFloat(e.currentTarget.getAttribute('x')), 
+                             y: parseFloat(e.currentTarget.getAttribute('y'))},
+                            'node');
+            } else {
+                findClosest( curElbow, 'elbow', 
+                            {x: parseFloat(e.currentTarget.getAttribute('x')), 
+                             y: parseFloat(e.currentTarget.getAttribute('y'))},
+                            'node'); 
+            }
+            select(e.currentTarget);
+              
             var thePath = document.createElementNS(xmlns, 'path');
             thePath.setAttributeNS(null, 'd', curPath);
             thePath.setAttributeNS(null, 'fill', 'none');
             thePath.setAttributeNS(null, 'stroke', 'black');
             thePath.setAttributeNS(null, 'data-active', 'drawn');
-            // thePath.setAttributeNS(null, 'marker-end', 'url(#arrow)');
+            thePath.setAttributeNS(null, 'marker-end', 'url(#arrow)');
             thePath.addEventListener('click', pathClicked, false);
             console.log('creating it ' + curPath);
             document.getElementById('mySVG').appendChild(thePath);
 
-            thePath.attributes['start'] = startNode;
-            thePath.attributes['end'] = e.currentTarget;
+            thePath.attributes['parents'] = startNode;
+            thePath.attributes['children'] = e.currentTarget;
 
             // update relationships
             startNode.attributes['children'].push(e.currentTarget);
             e.currentTarget.attributes['parents'].push(startNode);
             startNode.attributes['outEdges'].push(thePath);
             e.currentTarget.attributes['inEdges'].push(thePath);
-            // startNode.children
             startNode = null;
             curPath = null;
+            curElbow = null;
+        }
+    }
+}
+
+
+/*
+Return the best coordinates for the start and end of a edge. theNode and end could
+be a node or an elbow. Atleast one of beg and end must be a node.
+*/
+function findClosest(beg, typeB, end, typeE) {
+    'use-strict'
```

Comment:
it looks like a couple of functions are missing semi-colons after the `'use-strict'` statement.